utils/gitlab_api.py

```diff
@@ -293,22 +297,42 @@ def get_access_level(access):
         elif access == "guest":
             return gitlab.GUEST_ACCESS
 
+    def get_group_id_and_projects(self, group_name):
+        groups = self.gl.groups.list()
+        group = [g for g in groups if g.path == group_name]
+        if len(group) != 1:
+            logging.error(group_name + " group not found")
+            return []
+        [group] = group
```

Comment:
Now you're just removing the list... maybe it's not worth it to use the comprehension in the first place...

Reply:
is there a better/cleaner way to get a single element from a list based on a condition?

Reply:
My point is that maybe you should not have a list in the first place. Why to create a list of one group?

Reply:
It seems like the fact you hace a list is just a consequence of using the list comprehension on line 302. A regular for would avoid that.

Reply:
as discussed, we'll keep the current implementation for now